### PR TITLE
Chain navigation

### PR DESCRIPTION
**Pull request by gemdev202**
[Screen_recording_20241128_024220.webm](https://github.com/user-attachments/assets/1d7919b6-4a21-4680-b884-0c751554fea9)

https://developer.android.com/guide/navigation/backstack/circular

### CODE REVIEW

File: app/src/main/java/com/gemwallet/android/features/asset/details/models/AssetInfoUIModel.kt [hunk 1]
```diff
@@ -16,6 +17,7 @@ class AssetInfoUIModel(
     val tokenType: AssetType = AssetType.NATIVE,
     val networkTitle: String = "",
     val networkIcon: String = "",
+    val chain: Chain,
```
Comment (furenster) [inline app/src/main/java/com/gemwallet/android/features/asset/details/models/AssetInfoUIModel.kt:20]: Could you use asset for it?

File: app/src/main/java/com/gemwallet/android/features/asset/details/views/AssetDetailsScene.kt [hunk 2]
```diff
@@ -78,6 +78,7 @@ fun AssetDetailsScene(
     onSwap: (AssetId, AssetId?) -> Unit,
     onTransaction: (txId: String) -> Unit,
     onChart: (AssetId) -> Unit,
+    onChain: (AssetId) -> Unit,
```
Comment (furenster) [inline app/src/main/java/com/gemwallet/android/features/asset/details/views/AssetDetailsScene.kt:81]: Bad name:
- you could use it for any asset
- should use AssetIdAction

Comment (gemdev202) [reply app/src/main/java/com/gemwallet/android/features/asset/details/views/AssetDetailsScene.kt]: renamed to openNetwork with AssetIdAction

File: app/src/main/java/com/gemwallet/android/features/asset/navigation/AssetNavigation.kt [hunk 3]
```diff
@@ -18,8 +17,11 @@ internal const val assetIdArg = "assetId"
 const val assetRoute = "asset"
 
 fun NavController.navigateToAssetScreen(assetId: AssetId, navOptions: NavOptions? = null) {
-    navigate("$assetRoute/${assetId.toIdentifier().urlEncode()}", navOptions  ?: navOptions {
-        launchSingleTop = true
+    val route = "$assetRoute/${assetId.toIdentifier().urlEncode()}"
+    navigate(route, navOptions  ?: androidx.navigation.navOptions {
+        popUpTo(route) {
```
Comment (furenster) [inline app/src/main/java/com/gemwallet/android/features/asset/navigation/AssetNavigation.kt:22]: Did you check double tap?

**Review by furenster — COMMENTED** [review]
Check comments

File: app/src/main/java/com/gemwallet/android/features/asset/navigation/AssetNavigation.kt (hunk 3 above)
Comment (gemdev202) [reply app/src/main/java/com/gemwallet/android/features/asset/navigation/AssetNavigation.kt:22]: yes it open only single screen

File: app/src/main/java/com/gemwallet/android/features/asset/details/models/AssetInfoUIModel.kt [hunk 4]
```diff
@@ -15,6 +16,7 @@ class AssetInfoUIModel(
     val priceChangedType: PriceState = PriceState.Up,
     val tokenType: AssetType = AssetType.NATIVE,
     val networkTitle: String = "",
+    val networkIcon: String = "",
```
Comment (furenster) [inline app/src/main/java/com/gemwallet/android/features/asset/details/models/AssetInfoUIModel.kt:19]: Is it used?

**Pull request merged**